llvm/lib/Target/AIE/AIEPostPipeliner.cpp

```diff
@@ -270,8 +270,15 @@ bool PostPipeliner::scheduleFirstIteration() {
     LLVM_DEBUG(dbgs() << "  Emit in " << -Depth + LocalCycle << "\n");
     int Cycle = -Depth + LocalCycle;
     LLVM_DEBUG(dbgs() << "  Emit in " << Cycle << "\n");
-    HR.emitInScoreboard(Scoreboard, MI->getDesc(), MemoryBanks, MI->operands(),
-                        MI->getMF()->getRegInfo(), Cycle);
+    for (int N = 0; N < NCopies; N++) {
+      if (N > 0 && HR.checkConflict(Scoreboard, *MI, Cycle)) {
+        return false;
+      }
+
+      HR.emitInScoreboard(Scoreboard, MI->getDesc(), MemoryBanks,
+                          MI->operands(), MI->getMF()->getRegInfo(), Cycle);
+      Cycle += II;
+    }
```

Comment:
Noice!

Reply:
Could you think of maybe adding a small test that benefits from "pre-booking modulo resources"?

Reply:
I'll see whether this tipped the conv2d balamce.